### DISCUSSION TITLE
Refactoring and Tests

### DISCUSSION THREAD

**Discussion by Serene-Arc** [Neurrone/beets-audible#37] [Ideas]
Hi, I've made a couple of PRs before and I use this tool a lot to organise my audiobook library which is extensive. I was wondering if you've given any thought to tests to allow for more easy changes. This would require putting the plugin files in their own folder which would be any easy refactor and I think it would add a lot to the project moving forwards.

I'm happy to start this as I go about the code if you'd like. I can start a new branch and PR that details the progress of writing tests in draft mode so you can see the new code and offer feedback.

**Comment by Neurrone** [comment]
Thanks for offering to help with this, that would really be appreciated.

I currently don't have the time to do this myself but have always had this on my todo list, so any help would be most welcome.

**Reply by Serene-Arc** [reply]
Great :) I'll begin soon, I'd like to narrow down behaviour vis-a-vis sorting tracks and that would be a lot easy with a test base to check against.

**Reply by Neurrone** [reply]
That would be a great place to start. That's probably one of the more complex parts of the code which can be easily tested.

**Reply by Serene-Arc** [reply]
Also, would you like me to format according to PEP8 as I go? It's a small thing but it would help keep things consistent moving forwards.

**Reply by Neurrone** [reply]
I've been meaning to format the whole repository so that the formatting is standardized. Would you mind doing the formatting in a separate PR for ease of review?

**Reply by Neurrone** [reply]
I should have time to work on this more in a couple of weeks. I plan to set up stuff like a proper CI pipeline via GitHub actions which would also help with standardizing formatting or running tests.

**Reply by Serene-Arc** [reply]
Sure, I've formatted and I'll push the PR now.
